src/background/services/openPayments.ts

```diff
@@ -501,13 +501,23 @@ export class OpenPaymentsService {
     // correctly (either specific grant or all grants). See
     // https://github.com/interledger/web-monetization-extension/pull/379#discussion_r1660447849
     const grant = recurringGrant || oneTimeGrant
-
-    if (grant) {
-      await this.client!.grant.cancel(grant.continue)
-      await this.storage.clear()
-      this.grant = null
-      this.token = { value: '', manageUrl: '' }
+    if (!grant) {
+      return
     }
+
+    await this.client!.grant.cancel(grant.continue).catch((error) => {
+      if (error instanceof OpenPaymentsClientError) {
+        if (error.status === 400 && error.code === 'invalid_client') {
+          // key already removed from wallet
+          return
```

Comment:
Nit: went with `.catch` instead of `try {} catch {}` as can return easily here.

Reply:
In this case, if the key is revoked and the user wants to disconnect their wallet we should clear the storage. The user can revoke the grant in the wallet UI as well.

Reply:
Which is what it's doing here? To be clear, we're ignoring the error only with this return here - it's not returning from `disconnectWallet()` here, so we still clear the storage.

Reply:
My bad, I did not notice `.catch()`.